cmake/wasi-sdk-sysroot.cmake

```diff
@@ -243,9 +243,17 @@ endforeach()
 # misc build logic
 # =============================================================================
 
-install(DIRECTORY ${wasi_tmp_install}/lib ${wasi_tmp_install}/share
+install(DIRECTORY ${wasi_tmp_install}/share
         USE_SOURCE_PERMISSIONS
         DESTINATION ${CMAKE_INSTALL_PREFIX})
+cmake_path(IS_PREFIX CMAKE_INSTALL_PREFIX ${clang_resource_dir} NORMALIZE install_resource_dir)
+if(install_resource_dir)
+  install(DIRECTORY ${wasi_resource_dir}/lib
+          USE_SOURCE_PERMISSIONS
+          DESTINATION ${clang_resource_dir})
+else()
+  message(STATUS "The resource dir (${clang_resource_dir}) will not be updated by the install target because it's out of CMAKE_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX}")
+endif()
```

Comment:
This to me seems odd to me to have a conditional installation. Could an option be added to modify the host compiler's resource-dir and if that's turned off then the preexisting install location is used? (or something similar to it, probalby not the exact same structure with a clang version that isn't guaranteed to match something)

Reply:
i'm a bit confused because i just tried to follow your suggestion and you are now saying it's odd. :-)

how about the last push?

Reply:
The first version of this PR unconditionally modified the host compiler resource-dir which I felt was surprising since it was modifying things outside of `CMAKE_INSTALL_PREFIX`. The second version of this PR, which this comment was targetting, did nothing by default and happened to install things if `CMAKE_INSTALL_PREFIX` lined up. I found that surprising as well because it's not documented and just a status message can be easily lost in configure output.

Third version of this PR makes more sense. Something is always installed and it always shows up somewhere, and an opt-in is enabled to perform a non-standard installation to put something outside of the installation prefix.